README
NAME

    Dist::Zilla::PluginBundle::ACPS - Basic plugins to maintain and release
    ACPS dists

VERSION

    version 0.29

DESCRIPTION

    Plugin bundle for creating and maintaining Perl distributions for ACPS.
    It is equivalent to this:

     [GatherDir]
     [PruneCruft]
     [ManifestSkip]
     [MetaYAML]
     [MetaJSON]
     [License]
     [Readme]
     [ExecDir]
     [ModuleBuild]
     [Manifest]
     [TestRelease]
     [ConfirmRelease]
     
     [PodWeaver]
     [NextRelease]
     [AutoPrereqs]
     [OurPkgVersion]
     
     [Git::Check]
     [ACPS::Git::Commit]
     [ACPS::Release]

AUTHOR

    Graham Ollis <gollis@sesda3.com>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2012 by NASA GSFC.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for {{$dist->name}}

{{$NEXT}}

0.31      2015-09-24 15:16:10 America/New_York
  - dzil apbuild now creates a per user filelist

0.30      2015-01-06 09:14:11 America/New_York
  - bump version to reintroduce to CPAN

0.29      2014-12-29 12:47:05 America/New_York
  - miniting profile Clustericious is now optional, as it has been removed
    from CPAN

0.28      2014-04-07 11:47:43 America/New_York
  - [@ACPS] mb_ prefixed values are passed into [ModuleBuild], if present

0.27      2013-07-06 18:07:02 America/New_York
  - commit README.pod after release if it is there

0.26      2013-06-19 17:09:39 America/New_York
  - don't generate debug packages when building rpms

0.25      2013-06-14 17:25:03 America/New_York
  - don't prune .travis.yml from dist

0.24      2013-06-12 09:47:26 America/New_York
  - modified default dist.ini (comments only)

0.23      2013-06-11 07:31:47 America/New_York
  - use [Author::Plicease::UploadToCPAN]

0.22      2013-06-11 07:30:03 America/New_York
  - add .travis.yml to minted dist

0.21      2013-05-25 23:22:24 America/New_York
  - Allow README.pod to be dirty for release

0.20      2013-05-17 20:19:57 America/New_York
  - Allow main repo to be set via environment
  - documentation tweaks

0.18      2013-05-07 13:20:20 EST5EDT
  - remove [ExtraTests] which we don't use

0.17      2013-05-02 13:01:11 EST5EDT
  - require [MinimumPerl]

0.16      2013-04-23 16:35:01 America/New_York
  - created [@ACPS::MakeMaker]
  - fixed use test

0.15      2013-03-01 13:58:47 America/New_York
  - automatically update Changes after release

0.14      2013-03-01 13:38:40 America/New_York
  - move commands into its own dist

0.13      2013-03-01 13:26:11 America/New_York
  - import File::ShareDir( dist_dir) where we need it

0.11      2013-03-01 11:06:47 America/New_York
  - rm [MakeMaker] from [@ACPS]
  - rm [ShareDir] from [@ACPS]
  - mingint profiles for clustericiousa app, client and mbd_app

0.10      2013-02-22 14:51:32 America/New_York
  - commit to release instead of release/master
  - removed [ACPS::JavaScriptVersion]
  - default profile to include [RemovePrereqs], [MinimumPerl] and commented 
    out [Prereqs] and [ACPS::RPM] by default

0.09      2012-12-11 12:20:22 America/New_York
  - added [ACPS::JavaScriptVersion]

0.08      2012-10-09 10:11:31 America/New_York
  - for [@ACPS::Legacy] / [ACPS::Legacy] use prereqs from META.json instead of detect

0.07      2012-09-24 11:24:37 America/New_York
  - added dzil apupload command

0.06      2012-08-22 10:43:06 America/New_York
  - added ACPS::RPM and dzil rpmspec command
  - @ACPS::Legacy and ACPS::Legacy for non Dist::Zilla aware dists
  - create .dzil/config.ini if it doesn't already exist.
  - print out useful diagnostic when dzil rpmspec is run without the RPM plugin
  - use Template instead of Text::Template for spec template.
  - Forked some [Git::*] components to ensure that commit/tags happen at 
    release time and push happens after release, so that if a push fails
    during a release doesn't muck everything else up.
  - [ACPS::RPM] is no longer a subclass of [RPM]

0.05      2012-07-26 12:48:53 America/New_York
  - fix some deps
  - basic rpm support

0.04      2012-07-25 16:20:49 America/New_York
  - add @Git back in
  - add VERSION to each module's POD
  - added ACPS minting profile.

0.03      2012-07-25 13:04:59 America/New_York
  - self hosting
  - remove git from bundle
  - automatically create remote public
  - fail to release if there is already a tag for this version

0.02      2012-07-25 11:57:01 America/New_York
  - added TODO

0.01      2012-07-25 11:47:41 America/New_York
  - initial version


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Basic plugins to maintain and release ACPS dists",
  "dist_author" => [
    "Graham Ollis <gollis\@sesda3.com>"
  ],
  "dist_name" => "Dist-Zilla-PluginBundle-ACPS",
  "dist_version" => "0.29",
  "license" => "perl",
  "module_name" => "Dist::Zilla::PluginBundle::ACPS",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Cwd" => 0,
    "DateTime" => 0,
    "Dist::Zilla" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::AfterMint" => 0,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::BuildPL" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::Git::DirtyFiles" => 0,
    "Dist::Zilla::Role::Git::Repo" => 0,
    "Dist::Zilla::Role::MintingProfile" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "File::Copy" => 0,
    "File::HomeDir" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "File::Temp" => 0,
    "File::chdir" => 0,
    "Git::Wrapper" => "0.021",
    "IPC::Open3" => 0,
    "IPC::System::Simple" => 0,
    "JSON" => 0,
    "List::MoreUtils" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "MooseX::Has::Sugar" => 0,
    "MooseX::Types::Moose" => 0,
    "MooseX::Types::Path::Class" => 0,
    "Path::Class" => 0,
    "Path::Class::Dir" => 0,
    "String::Formatter" => 0,
    "Template" => 0,
    "Try::Tiny" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "perl" => "v5.10.0",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "share_dir" => {
    "dist" => "share"
  },
  "test_requires" => {
    "FindBin" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  }
);


my %fallback_build_requires = (
  "FindBin" => 0,
  "Module::Build" => "0.3601",
  "Test::DZil" => 0,
  "Test::More" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share module => "Dist::Zilla::MintingProfile::ACPS", "profiles";


my %WriteMakefileArgs = (
  "ABSTRACT" => "the basic plugins to maintain and release ACPS dists",
  "AUTHOR" => "Graham Ollis <gollis\@sesda3.com>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601",
    "Test::More" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.03",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-ACPS",
  "EXE_FILES" => [],
  "LICENSE" => "restrictive",
  "NAME" => "Dist::Zilla::PluginBundle::ACPS",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Cwd" => 0,
    "DateTime" => 0,
    "Dist::Zilla" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::AfterMint" => 0,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::BuildPL" => 0,
    "Dist::Zilla::Role::Git::DirtyFiles" => 0,
    "Dist::Zilla::Role::Git::Repo" => 0,
    "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "File::Copy" => 0,
    "File::HomeDir" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "File::Temp" => 0,
    "File::chdir" => 0,
    "Git::Wrapper" => "0.021",
    "IPC::Open3" => 0,
    "IPC::System::Simple" => 0,
    "JSON" => 0,
    "List::MoreUtils" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "MooseX::Has::Sugar" => 0,
    "MooseX::Types::Moose" => 0,
    "MooseX::Types::Path::Class" => 0,
    "Path::Class" => 0,
    "Path::Class::Dir" => 0,
    "String::Formatter" => 0,
    "Template" => 0,
    "Try::Tiny" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.10",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


